components/global_header/right_controls/cloud_upgrade_button/index.tsx

```diff
@@ -39,9 +39,17 @@ const UpgradeCloudButton = (): JSX.Element | null => {
     const dispatch = useDispatch();
     const {formatMessage} = useIntl();
 
+    const isCloudFreeEnabled = useSelector(cloudFreeEnabled);
+    const isAdmin = useSelector((state: GlobalState) => isCurrentUserSystemAdmin(state));
+    const subscription = useSelector(selectCloudSubscription);
+    const product = useSelector(selectCloudProduct);
+    const isCloud = useSelector(isCurrentLicenseCloud);
+
     useEffect(() => {
-        dispatch(getCloudSubscription());
-        dispatch(getCloudProducts());
+        if (isCloud) {
+            dispatch(getCloudSubscription());
+            dispatch(getCloudProducts());
+        }
     }, []);
```

Comment:
`isCloud` should be in the deps array. It will still avoid unnecessary calls if it is, because it is only called when the component first renders or when the boolean value changes, and it should only change from `false` to `true` at most once as the data it is based on loads.